src/app/queries/fetchHelpers.ts

```diff
@@ -20,7 +20,10 @@ export const useFetchContext = (): IFetchContext => {
 export const authorizedFetch = async <T>(
   url: string,
   fetchContext: IFetchContext,
-  extraHeaders: RequestInit['headers'] = {}
+  extraHeaders: RequestInit['headers'] = {},
+  method: 'get' | 'post' = 'get',
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  data?: any // TODO: type this
```

Comment:
I think maybe the correct way to type this is to have it be another generic type param that defaults to unknown, e.g. instead of `<T>` we could have `<TResponse, TData = unknown>`. That way if we really need to we can restrict the type of the object we pass into a particular call. That's how react-query handles this type of thing internally.

If TS doesn't like passing `unknown` into `JSON.stringify()` we could use `any` as the default instead, but it'd be nice to not have to disable eslint here. And this feels like a good use case for `unknown`... it's not that the param can be anything, it's that this function doesn't care what it is.